openstack/messaging/v2/queues/results.go

```diff
@@ -77,13 +82,34 @@ type QueueDetails struct {
 	Flavor string `json:"flavor"`
 }
 
+// Stats represents a stats response.
+type Stats struct {
+	// Number of Claimed messages for a queue
+	Claimed int `json:"claimed"`
+
+	// Total Messages for a queue
+	Total int `json:"total"`
+
+	// Number of free messages
+	Free int `json:"free"`
+}
+
 // Extract interprets any commonResult as a Queue.
 func (r commonResult) Extract() (QueueDetails, error) {
 	var s QueueDetails
 	err := r.ExtractInto(&s)
 	return s, err
 }
 
+// Extract interprets any StatResult as a Stats.
+func (r StatResult) Extract() (Stats, error) {
+	var s struct {
+		Messages Stats `json:"messages"`
```

Comment:
```go
Stats Stats `json:"messages"`
```

`Messages` could infer actual messages would be returned. I understand `messages` is the JSON tag, though.